lampod/src/ln/peer_manager.rs

```diff
@@ -141,30 +141,32 @@ impl LampoPeerManager {
         Ok(())
     }
 
-    #[allow(unused_must_use)]
     pub fn run(&self) -> error::Result<()> {
         let listen_port = self.conf.port;
         let Some(ref peer_manager) = self.peer_manager else {
             error::bail!("peer manager is None, at this point this should be not None");
         };
         let peer_manager = peer_manager.clone();
         std::thread::spawn(move || {
-            async_run!(async move {
+            let result = async_run!(async move {
```

Comment:
nit: this is hiding a huge bug, you never return the error to the caller

```suggestion
            let _ = async_run!(async move {
```